File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/misc/SinglePressButton.java
package org.firstinspires.ftc.teamcode.misc;

import java.util.function.BooleanSupplier;

public class SinglePressButton {

    private final BooleanSupplier booleanSupplier;
    private boolean lastButtonState = false;

    public SinglePressButton(BooleanSupplier booleanSupplier) {
        this.booleanSupplier = booleanSupplier;
    }


    public boolean get() {
        boolean currentButtonState = booleanSupplier.getAsBoolean();
        boolean trigger = ((currentButtonState != lastButtonState) && currentButtonState);
        lastButtonState = currentButtonState;
        return trigger;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/misc/SinglePressButton.kt
package org.firstinspires.ftc.teamcode.misc

import java.util.function.BooleanSupplier

class SinglePressButton(private val booleanSupplier: BooleanSupplier) {
    private var lastButtonState = false
    fun get(): Boolean {
        val currentButtonState = booleanSupplier.asBoolean
        val trigger = currentButtonState != lastButtonState && currentButtonState
        lastButtonState = currentButtonState
        return trigger
    }
}